### DISCUSSION TITLE
What are you building with mobx-keystone?

### DISCUSSION THREAD

**Discussion by justin-hackin** [xaviergonz/mobx-keystone#321] [General]
Curious to know what the community is doing with this library, would appreciate hearing more about your experience.

I'll shamelessly kick off this discussion by sharing the app I just published after porting from mobx-state-tree to mobx-keystone: [Polyhedz Studio](https://github.com/justin-hackin/polyhedz-studio). 

I'm very pleased with what I was able to achieve with it, especially the concept of "[tweakable](https://github.com/justin-hackin/polyhedz-studio/tree/master/packages/renderer/src/common/keystone-tweakables)" nodes.

**Comment by knjname** [comment]
I'm using mobx-keystone for a new B2B web service launched in my company. It's already on production.
It covers various features from manipulating drawings as in PowerPoint to operating admin panels.
In any case, I can code complicated SPA logics in intuitive way.

I love mobx-keystone's TypeScript-native types and fast compile time.
(In my worst case, MST takes 1 hour with complex inferred types.)

I also like context feature. Referencing values between nodes was always problematic.
Context is a nice way to solve it.

**Comment by maccman** [comment]
a note taking tool: https://reflect.app !

**Reply by quolpr** [reply]
Actually we are building the similar tools, and it's funny, that you came to mobx-keystone too 😅

**Comment by quolpr** [comment]
A note-taking tool too 😅 https://github.com/quolpr/harika

Offline first with SQL.js storage + mobx-keystone as a state manager, that partially loads notes data from SQL.js

**Reply by quolpr** [reply]
I needed somehow to track changes in any piece of data. That small changes were dumping to SQL.js storage.

I tried redux, and it's hard to track changes with it — you need to make diffs on every action apply, which has a huge performance impact. 
I also tried MST, but typescript support was not so good

The mobx-keystone is just a diamond among all. It is soo comfortable to use and perfectly fits my requirements. Nothing else is close to it

**Reply by quolpr** [reply]
And it's sad that mobx-keystone is not so popular as it should be. I repeat myself, it's perfect 😄 It's better than the other state managers I ever saw

**Reply by hcschuetz** [reply]
Years ago, in the Java world, I have worked with EMF (Eclipse Modelling Framework).  Finally I have found mobx-keystone as something comparable in the JS/TS world.

**Comment by hcschuetz** [comment]
Recently I have been looking around for alternatives to React, basically out of curiosity. One thing I came up with was using plain web components **without React** but with reactive state based on mobx/mobx-keystone.

As an experiment I have implemented the classical TodoMVC app in this way ([source](https://github.com/hcschuetz/todomvc-mobx-jsx)). The [demo](https://hcschuetz.github.io/todomvc-mobx-jsx/dist/) looks just like the usual TodoMVC and is only interesting if you want to use your browser's DOM inspector.

I think the experiment was successful.  A few more utility functions (but not very many) would be needed to make this a general-purpose approach.

OTOH, one of my reasons for investigating react-less approaches was the weight of React in the bundled apps.  And now mobx or mobx-keystone also add significant weight. My guess is that the app could get more light-weight if vite/rollup could do better tree-shaking, but with heavily overloaded identifiers such as `observable` tree-shaking is hard.

Don't be surprised to see JSX syntax.  This is not React, but a simple homegrown JSX implementation with some reactivity support.

BTW, I have started with mobx-state-tree but switched to mobx-keystone for the better TypeScript integration.
